www/js/survey/multilabel/infinite_scroll_filters.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * The general structure of this code is that all the timeline information for
+ * a particular day is retrieved from the Timeline factory and put into the scope.
+ * For best performance, all data should be loaded into the in-memory timeline,
+ * and in addition to writing to storage, the data should be written to memory.
+ * All UI elements should only use $scope variables.
+ */
+
+import i18next from "i18next";
+
+const unlabeledCheck = (t) => {
+    return t.INPUTS
+         .map((inputType, index) => typeof t.userInput[inputType] === 'undefined')
+         .reduce((acc, val) => acc || val, false);
+}
+
+const invalidCheck = (t) => {
+    const retVal =
+        (typeof t.userInput['MODE'] !== 'undefined' && t.userInput['MODE'].value === 'pilot_ebike') &&
```

Comment:
This used to be `angular.isDefined(x)` and was replaced as `typeof x !== 'undefined'`

Instead, you can simply do `!x` (in all places this was done)

Why? Because if `x` was, for some reason, `null` or `false` or `NaN` or any other falsy value, `typeof x !== 'undefined'` would be true.

We should treat any falsy value, whether `undefined` or not, as invalid / 'not filled in'.

Reply:
Fixed this in: https://github.com/e-mission/e-mission-phone/pull/1062/commits/259778a0d994846693de0ec405e051f8ab09f040